specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2020-12-01/Marketplace.json

```diff
@@ -0,0 +1,1068 @@
+      "properties": {
+        "offerId": {
+          "description": "Gets or sets offer Id",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "Gets display name",
+          "type": "string",
+          "readOnly": true
+        },
+        "publisherId": {
+          "description": "Gets or sets publisher Id",
+          "type": "string"
+        },
+        "adminAction": {
+          "description": "Gets or sets admin action",
+          "type": "string",
+          "enum": [
+            "Approved",
+            "Rejected"
+          ],
+          "x-ms-enum": {
+            "name": "adminAction",
+            "modelAsString": true
+          }
+        },
+        "approvedPlans": {
+          "description": "Gets or sets Approved plans ids, empty in case of rejected",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "comment": {
+          "description": "Gets or sets admin comment",
+          "type": "string"
+        },
+        "administrator": {
+          "description": "Gets or sets admin details",
+          "type": "string"
+        },
+        "plans": {
+          "description": "Gets list of plans with requesters details",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/PlanRequesterDetails"
+          }
+        }
+      }
+    },
+    "PlanRequesterDetails": {
+      "description": "Plan with requesters details",
+      "properties": {
+        "planId": {
+          "description": "Gets the plan id",
+          "type": "string",
+          "readOnly": true
+        },
+        "planDisplayName": {
+          "description": "Gets the plan display name",
+          "type": "string",
+          "readOnly": true
+        },
+        "requesters": {
+          "description": "Gets requesters details list",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/UserRequestDetails"
+          }
+        }
+      }
+    },
+    "UserRequestDetails": {
+      "description": "user request details",
+      "type": "object",
+      "properties": {
+        "user": {
+          "description": "Gets user id",
+          "type": "string",
+          "readOnly": true
+        },
+        "date": {
+          "description": "Gets request date",
+          "type": "string",
+          "readOnly": true
+        },
+        "justification": {
+          "description": "Gets justification",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RequestApprovalsList": {
+      "description": "List of admin request approval resources",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RequestApprovalResource"
+          }
+        },
+        "nextLink": {
```

Comment:
mark as readOnly because your service is setting this value.

Reply:
done